src/Arcus.Testing.Integration.DataFactory/TemporaryDataFlowDebugSession.cs

```diff
@@ -431,14 +431,74 @@ public async ValueTask DisposeAsync()
         {
             if (_startedByUs)
             {
-                _logger.LogDebug("[Test:Teardown] Stop Azure Data Factory '{Name}' data flow debug session '{SessionId}'", DataFactory.Id.Name, SessionId);
+                _logger.LogTeardownStopSession(DataFactory.Id.Name, SessionId);
                 await DataFactory.DeleteDataFlowDebugSessionAsync(new DeleteDataFlowDebugSessionContent { SessionId = SessionId });
             }
 
             GC.SuppressFinalize(this);
         }
     }
 
+    internal static partial class TempoDataFlowDebugSessionILoggerExtensions
+    {
+        private const LogLevel SetupTeardownLogLevel = LogLevel.Debug;
+
+        [LoggerMessage(
+            Level = LogLevel.Trace,
+            Message = "[Test:Setup] Starting Azure Data Factory '{DataFactoryName}' data flow debug session... (might take up to 3 min to start up)")]
+        internal static partial void LogSetupStartingSession(this ILogger logger, string dataFactoryName);
+
```

Comment:
TIL.
I wasn't aware of the [LoggerMessageAttribute](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggermessageattribute?view=net-9.0-pp). Thanks!